### PR TITLE
[MODORDERS-1015] - Implement endpoint to bind multiple pieces together(no items associated)

### PR DESCRIPTION
**Pull request by azizbekxm**
## Purpose

https://folio-org.atlassian.net/browse/MODORDERS-1015
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Examples exist for all schemas
  - [ ] Descriptions exist for all schema properties
  - [ ] All schemas pass raml-cop linting
- Does this introduce breaking changes?
  - [ ] Were there any schema changes?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: mod-orders/examples/bindPiecesCollection.sample [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "toBeBound": [
+    {
+      "poLineId": "3f4f61ab-d3c2-47a9-baa6-c903a4cfba57",
+      "bound": 2,
+      "bindPieces": [
+        {
+          "id": "fe729e0b-5c56-4f0a-bb3a-33bc405d71a0",
+          "comment": "Captivating cover art for the first edition"
```
Comment (azizbekxm) [inline mod-orders/examples/bindPiecesCollection.sample:9]: Comment probably is not required, will be deleted

File: mod-orders/examples/bindPiecesCollection.sample [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+  "toBeBound": [
```
Comment (SerhiiNosko) [inline mod-orders/examples/bindPiecesCollection.sample:2]: Good that we follow an existing approach of payload like to beReceieved etc
I also think that bound, totalRecords are redundant. and in previous payloads as well.
If we remove these 2 fields we will make this payload inconsistent with other similar, so let's keep

Comment (SerhiiNosko) [inline mod-orders/examples/bindPiecesCollection.sample:2]: We also need add these 5 fields to be able to create new item and connect pieces to that item
image

Comment (SerhiiNosko) [inline mod-orders/examples/bindPiecesCollection.sample:2]: In the AC of releated UI story written
image
And the same title has the same poLined so we don't need possibility to specify multiple poLineIds at the same time.
Need to think how to simplify the contract

Comment (SerhiiNosko) [reply mod-orders/examples/bindPiecesCollection.sample:2]: other services use the same structure, so you can ignore this comment

Comment (azizbekxm) [reply mod-orders/examples/bindPiecesCollection.sample:2]: thanks, added extra 5 fields along with poLineId

Comment (azizbekxm) [reply mod-orders/examples/bindPiecesCollection.sample:2]: Ok, thanks to clarification

**Review by SerhiiNosko — APPROVED** [review]

File: mod-orders/examples/bindPiecesCollection.sample [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  "toBeBound": {
+    "poLineId": "3f4f61ab-d3c2-47a9-baa6-c903a4cfba57",
+    "bindItem": {
+      "barcode": "0987654111",
+      "materialTypeId": "a058cb27-a77d-4cb3-b76b-2adc931fb0c9",
+      "callNumber": "BF2050 .M335 2000",
+      "permanentLoanTypeId": "41673a86-ebab-4a5a-8b1b-76110a6f86f4",
+      "permanentLocationId": "02cc63e9-dda0-4b49-bbc6-6253edffcc67"
+    },
+    "bound": 2,
```
Comment (SerhiiNosko) [inline mod-orders/examples/bindPiecesCollection.sample:11]: lets delete bound and totalRecords in this case

Comment (azizbekxm) [reply mod-orders/examples/bindPiecesCollection.sample]: deleted

**Pull request merged**